R/Predict.R

```diff
@@ -75,24 +75,32 @@ Predict <-
     }
 
     if (!aggregate.output) {
-      predictions <- matrix(forest$labels[unlist(Yhats)], nrow(X), f_size)
+      if (task == 'classificaiton')
+        predictions <- matrix(forest$labels[unlist(Yhats)], nrow(X), f_size)
+      else
+        predictions <- matrix(unlist(Yhats), nrow(X), f_size)
     } else {
-      num_classes <- ncol(forest$trees[[1L]]$ClassProb)
-      predictions <- matrix(0L,nrow=nrow(X), ncol=num_classes)
-      for (m in 1L:f_size) {
-        for (k in 1L:nrow(X)) {
-          predictions[k, Yhats[[m]][k]] <- predictions[k, Yhats[[m]][k]] + 1L
+      if (task == 'classification') {
+        num_classes <- ncol(forest$trees[[1L]]$ClassProb)
+        predictions <- matrix(0L,nrow=nrow(X), ncol=num_classes)
+        for (m in 1L:f_size) {
+          for (k in 1L:nrow(X)) {
+            predictions[k, Yhats[[m]][k]] <- predictions[k, Yhats[[m]][k]] + 1L
+          }
         }
-      }
-      predictions <- predictions/f_size
-      if (!output.scores) {
-        if (is.integer(forest$labels)) {
-          predictions <- forest$labels[max.col(predictions)]
-        } else {
-          predictions <- factor(forest$labels[max.col(predictions)], levels = forest$labels)
+        predictions <- predictions/f_size
+        if (!output.scores) {
+          if (is.integer(forest$labels)) {
+            predictions <- forest$labels[max.col(predictions)]
+          } else {
+            predictions <- factor(forest$labels[max.col(predictions)], levels = forest$labels)
+          }
         }
+      } else {
```

Comment:
we probably want to change this to else if to be clear about what's going on

Reply:
I added a check at that beginning of the function to check the task is either classification or regression. The design of the if-else statements is very bad. Changing every one of them to be tryCatch is not good design. It should be abstracted into a single class and methods should be overloaded based on whether regression or classification is used.